configure.ac

```diff
@@ -0,0 +1,417 @@
+AS_IF([test "x${ac_cv_func_logwtmp}" != "xyes"], [
+    AC_CHECK_LIB([util], [logwtmp], [
+        AC_DEFINE(HAVE_LOGWTMP, 1, ["System provides the logwtmp() function"])
+        AC_SUBST([UTIL_LIBS], ["-lutil"])
+    ])
+])
+
+#
+# Check if libcrypt have crypt() function
+AC_CHECK_LIB([crypt], [crypt],
+    AC_SUBST([CRYPT_LIBS], ["-lcrypt"]))
+
+#
+# Should pppd link with -lsystemd (Linux only)
+AC_ARG_ENABLE([systemd],
+    AS_HELP_STRING([--enable-systemd], [Enable support for systemd notification]))
+AM_CONDITIONAL(WITH_SYSTEMD, test "x${enable_systemd}" = "xyes")
+AM_COND_IF([WITH_SYSTEMD],
+    AC_DEFINE([SYSTEMD], 1, [Enable support for systemd notifications]))
+
+#
+# Enable Callback Protocol Support, disabled by default
+AC_ARG_ENABLE([cbcp],
+    AS_HELP_STRING([--enable-cbcp], [Enable Callback Protocol]))
+AM_CONDITIONAL(WITH_CBCP, test "x${enable_cbcp}" = "xyes")
+AM_COND_IF([WITH_CBCP],
+    AC_DEFINE([CBCP_SUPPORT], 1, [Have Callback Protocol support]))
+
+#
+# Disable support for IPX control protocol
+AC_ARG_ENABLE([ipxcp],
+    AS_HELP_STRING([--enable-ipxcp], [Enable IPX Control Protocol support]))
+AM_CONDITIONAL(WITH_IPXCP, test "x${enable_ipxcp}" = "xyes")
+AM_COND_IF(WITH_IPXCP,
+    AC_DEFINE([IPX_CHANGE], 1, ["Have IPX Control Protocol"]))
+AS_IF([test "x${build_sunos}" = "xyes" && test "x${enable_ipxcp}" = "xyes"],
+    [AC_MSG_ERROR([IPXCP is not supported on SunOS, disable using --disable-ipxcp])])
+
+#
+# Disable support for limiting session duration by maximum octets
+AC_ARG_ENABLE([maxoctets],
+    AS_HELP_STRING([--disable-maxoctets], [Disable support for limiting session by maximum octets]))
+AS_IF([test "x$enable_maxoctets" != "xno"],
+    AC_DEFINE([MAXOCTETS], 1, ["Limit sessions by maximum number of octets"]))
+
+#
+# Disable Microsoft extensions will remove CHAP and MPPE support
+AC_ARG_ENABLE([microsoft-extensions],
+    AS_HELP_STRING([--disable-microsoft-extensions], [Disable Microsoft CHAP / MPPE extensions]))
+
+AM_CONDITIONAL(WITH_CHAPMS, test "x${enable_microsoft_extensions}" != "xno")
+AM_COND_IF([WITH_CHAPMS],
+    AC_DEFINE([CHAPMS], 1, ["Have Microsoft CHAP support"]))
+
+AM_CONDITIONAL(WITH_MPPE, test "x${enable_microsoft_extensions}" != "xno")
+AM_COND_IF([WITH_MPPE],
+    AC_DEFINE([MPPE], 1, ["Have Microsoft MPPE support"]))
+
+#
+# Enable Microsoft LAN Manager support, depends on Microsoft Extensions
+AC_ARG_ENABLE([mslanman],
+    AS_HELP_STRING([--enable-mslanman], [Enable Microsoft LAN Manager support]))
+AS_IF([test "x${enable_mslanman}" = "xyes" && test "x${enable_microsoft_extensions}" != "xno"],
+    AC_DEFINE([MSLANMAN], 1, ["Have Microsoft LAN Manager support"]))
+
+#
+# Disable IPv6 support
+AC_ARG_ENABLE([ipv6-support],
+    AS_HELP_STRING([--disable-ipv6-support], [Disable IPv6 support]))
+AM_CONDITIONAL(WITH_INET6, test "x${enable_ipv6_support}" != "xno")
+AM_COND_IF([WITH_INET6],
+    AC_DEFINE(INET6, 1, ["Have IPv6 support"]))
```

Comment:
... and IPv6 protocol support named `--enable-ipv6-support` with define `INET6`?

It looks like a big inconsistency.

In both cases you are enabling / disabling PPP control protocol (IPXCP and IPV6CP) and not support for particular protocol.

Technically speaking you can use IPX or IPv6 protocols over PPP also without corresponding control protocol in pppd. Just you need to handle stuff by your own way...

Reply:
@pali I was trying to be consistent with how these was defined in the previous configure scripts. The INET6 define does in fact disable IPv6 support entirely (and not just ipv6cp); The IPXCP is actually the IPXCP protocol. Grep for INET6 in the code ...

Reply:
> I was trying to be consistent with how these was defined in the previous configure scripts. 

Ok. (I was in impression that conversion to autotools means also cleanup of this stuff)

> The INET6 define does in fact disable IPv6 support entirely (and not just ipv6cp);

Apparently not. The only thing which pppd do with IPv6 is its Control Protocol. So _IPv6 entirely_ means IPV6CP protocol... Just code related to IPV6CP is not only in ipv6cp.c file but also in others...

Reply:
@pali You know that area of the code better than me. Am I understanding you correctly when you say that INET6 define actually enable/disables the IPV6CP? If so, then I do agree with you that the configure option for consistency would be better as --enable/disable-ipv6cp to match --enable-ipxcp.

On a slightly different note, all the #ifdef INET6 does have a risk of bleeding over to other projects as it is used in pppd's public headers as well. I'd rather change that to something along the lines of WITH_INET6, or WITH_PPP_IPV6CP. The latter would make it crystal clear that the feature belongs to pppd and not overlap with a third party's INET6 if they would define it for whatever reason. The same goes with other defines, e.g. CHAPMS, etc. 

While I was trying to avoid changing the defines because it is very intrusive to the source code. #296 was meant to mostly refactor the Makefiles / Build system while leaving the source code only with cosmetic fixes to it. The cleaner the fix is, the more likely it would be accepted and merged.

I'd be totally fine with changing the configure script to take AC_ARG_ENABLE([ipv6cp], ....) to match that of the of ipxcp option. What do you think? I can craft another fix for this...

Reply:
Also, before I head down that road, a thumbs up from @paulusmack would be helpful.

Reply:
I would say that there is something terribly wrong if there is a need to explicitly disable IPv6 at compile time in year 2021. Any kind of IPv6 support should not be disabled at compile time.

pppd for a long time can detect ppp link can do IPV6CP and disable its negotiation when other part does not support it. And in case other side is totally broken it is possible to disable IPV6CP at runtime by pppd option.

On the other hand, disabling IPX makes sense as there is probably nobody in year 2021 who uses IPX (and really nobody who has it over ppp).

Reply:
@pali that is probably true. I think there might be a use-case for custom embedded systems that doesn't have e.g. IPv6 compiled into the kernel. Do you suggest removing the define altogether? If so, I'd want to see @paulusmack input ...